docs/source/performance/cache-backends.mdx

```diff
@@ -0,0 +1,81 @@
+
+Unfortunately, the default configuration of Apollo Server exposes you to denial of service attacks.
+This is because APQs are enabled by default and the default cache in Apollo Server 3 is unbounded, meaning an attacker can exhaust your memory and crash your server. The default cache in Apollo Server 4 will be bounded; **we recommend opting in to that behavior by providing the `cache: "bounded"` option to your Apollo Server constructor or configuring the `cache` yourself.**
+
+## Configuring in-memory caching
+
+**Apollo Server's default caching features use an unbounded cache, which is not safe for production use.** If you'd like to configure the in-memory cache, Apollo provides the [`InMemoryLRUCache`](https://github.com/apollographql/apollo-utils/tree/main/packages/keyValueCache#inmemorylrucache) class from the `@apollo/utils.keyvaluecache` package.
+
+The `InMemoryLRUCache` class is a wrapper around the `lru-cache` package and has a default maximum of approximately 30MiB of memory. You can configure an instance of `InMemoryLRUCache` with the same options as found in the `lru-cache` package, see the [`lru-cache` documentation](https://www.npmjs.com/package/lru-cache) for more details. You can use Apollo's `InMemoryLRUCache` by passing it to the `cache` option of the `ApolloServer` constructor like so:
+
+```ts
+import { InMemoryLRUCache } from '@apollo/utils.keyvaluecache';
+
+const server = new ApolloServer({
+  // Note: this is equivalent to providing `cache: "bounded"`
+  cache: new InMemoryLRUCache(),
+});
+```
+
+In this example, we've increased the default size and provided a default TTL. For more information on these configuration options, see the [`lru-cache` documentation`](https://www.npmjs.com/package/lru-cache).
+```ts
+import { InMemoryLRUCache } from '@apollo/utils.keyvaluecache';
+
+const server = new ApolloServer({
+  // ...
+  cache: new InMemoryLRUCache({
+    // ~100MiB
+    maxSize: Math.pow(2, 20) * 100,
+    // 5 minutes (in milliseconds)
+    ttl: 300_000,
+  }),
+});
+```
+
+## Configuring external caching
+
+Apollo no longer maintains any caching backends directly. Instead, we recommend using the [`keyv`](https://www.npmjs.com/package/keyv) package along with the [`KeyvAdapter`](https://github.com/apollographql/apollo-utils/tree/main/packages/keyvAdapter#keyvadapter-class) class provided by the `@apollo/utils.keyvadapter` package. `KeyvAdapter` simply wraps a `Keyv` instance and implements the `KeyValueCache` interface which is required by Apollo Server. You can use the `KeyvAdapter` class to wrap a `Keyv` instance and provide it to the `cache` option of the `ApolloServer` constructor like so:
+
+```ts
```

Comment:
I do think we should provide complete examples for Redis and Memcached (replacing the old ones we have) which should include `npm install` commands: not obvious that you need to `npm install @keyv/redis`, eg.

Reply:
Let me know what you think: [909a913](https://github.com/apollographql/apollo-server/pull/6547/commits/909a913281eb58f7751178c14d51f72dff3d8ed0)